### DISCUSSION TITLE
Question - To have the Web portal on a separate page from the sync API?

### DISCUSSION THREAD

**Discussion by isudoi** [dani-garcia/vaultwarden#3253] [Q&A]
Question - To have the Web portal on a separate page from the sync API?

This is a question I had that came up while trying to secure my vaultwarden instance, since it's exposed to the internet

What I am trying to do is, since the mobile apps require the server url the same as the web portal for sync.

Is it possible to configure or maybe as a feature request to load the web portal on a different page than the API sync?

so I can secure the web portal with Cloudflare Tunnel as Application, and leave the API sync available on the internet?

for now, I cannot do for example

https://vaultwarden.domain.com/webportal

https://vaultwarden.domain.com

As both running the same:
https://vaultwarden.domain.com/webportal

Thanks

**Comment by sorcix** [comment]
There's nothing to protect on the web client. It's just a javascript web application, executed entirely client side. If I download the web app on my computer, execute it locally and point it to your API then I have exactly the same features. It makes no sense to hide or protect the web app. (Just disable the Vaultwarden admin page whe you're not using it.)

The API is the important part, so it looks like you're trying to do the exact opposite of what you'd need to do to get extra security.

**Reply by isudoi** [reply]
I understand that part, what I wanted to add is to protect the login page on the web browser with extra layer of login.

But currently, I can't do that since the API sync for mobile devices relies on the root level and no subpages to get the sync working.

Note: (Already have MFA enabled)

Note: the ADMIN page already protected with MFA Cloudflare tunnel.
https://your.domain.com/admin

Just wanted to see if there is any possible way to get the web portal similar to he admin page on a different page instead of the root level.

**Reply by sorcix** [reply]
You could host the web application using any webserver like nginx, apache or caddy.

The web vault for Vaultwarden can be found here: https://github.com/dani-garcia/bw_web_builds

But again: protecting the login page doesn't add any security. The actual login is done on your API. It makes no sense to protect the web app but not the API.

**Reply by isudoi** [reply]
@sorcix that would be a good idea and some nice project to do, and host it on a webserver with even a different subdomain/page! Thanks for sharing that.

Any idea how to disable the web page then on the default installation? and to keep the API functioning?

**Reply by BlackDex** [reply]
See https://github.com/dani-garcia/vaultwarden/blob/1ba8275dcb53bf8e6f41ad2fa2e0a09f3674ff74/.env.template#L66

**Reply by isudoi** [reply]
Found it here the docker compose env! it works! awesome!
https://github.com/dani-garcia/vaultwarden/wiki/Disabling-or-overriding-the-Vault-interface-hosting
Now, I need to figure out where to find and how to move the static files web vault into another webserver.

**Reply by isudoi** [reply]
I figured out the trick.

api sync already has its api endpoint url.
https://github.com/dani-garcia/vaultwarden/wiki/Using-an-alternate-base-dir
I allowed that through the proxy /api/sync

and then the root dir base is protected now behind a Cloudflrare tunnel portal!

Issue fixed!

Thanks all for the brainstorming
